### PR TITLE
General documentation improvements

### PR DESCRIPTION
**Pull request by CarsonHoffman**
I went through most of the major structs and threw some comments in, especially in relation to odd things present in the API which aren't fully reflected in the library. I tried my best to scan for typos, but if I ended up letting one through, oops.

### CODE REVIEW

**Comment by jai-sharma** [comment]
LGTM

**Comment by CarsonHoffman** [comment]
After looking at it again I went through and fixed a few nits, as well as squashing the commits into one.

File: user.go [hunk 1]
```diff
@@ -7,15 +7,35 @@ import (
...
 // A User stores all data for an individual Discord user.
 type User struct {
-	ID            string `json:"id"`
-	Email         string `json:"email"`
-	Username      string `json:"username"`
-	Avatar        string `json:"avatar"`
+	// The ID of the user.
+	ID string `json:"id"`
+
+	// The email of the user. This is only presence when
```
Comment (sn0w) [inline user.go]: presence -> present

**Comment by CarsonHoffman** [comment]
@sn0w thanks for the heads up; don't really know how I missed that. Hey, at least I'm finally learning how to squash commits today. :D

**Comment by bwmarrin** [comment]
This is fantastic! Thanks for putting work into documentation, I love it.

**Pull request merged**